packages/@ourworldindata/grapher/src/modal/DownloadModal.tsx

```diff
@@ -337,92 +375,487 @@ export class DownloadModal extends React.Component<DownloadModalProps> {
+        attributions[0].toLowerCase() === "our world in data"
+    const processingLevelPhrase = !sourceIsOwid
+        ? getPhraseForProcessingLevel(owidProcessingLevel)
+        : undefined
+    const fullProcessingPhrase = processingLevelPhrase ? (
+        <>
+            {" "}
+            – <i>{processingLevelPhrase} by Our World In Data</i>
+        </>
+    ) : undefined
+
+    return (
+        <div className="download-modal__data-section download-modal__sources">
+            <h3 className="grapher_h3-semibold">Source and citation</h3>
+            {sourceLinks.length > 0 && (
+                <div className="download-modal__data-sources">
+                    <strong>Data sources:</strong>{" "}
+                    <ul className="download-modal__data-sources-list">
+                        {sourceLinks}
+                    </ul>
+                    {fullProcessingPhrase}
+                </div>
+            )}
+            <div>
+                <strong>Citation guidance:</strong> Please credit all sources
+                listed above. Data provided by third-party sources through Our
+                World in Data remains subject to the original{" "}
+                {sourceLinks.length === 1 ? "provider's" : "providers'"} license
+                terms.
+            </div>
+        </div>
+    )
+}
+
+const ApiAndCodeExamplesSection = (props: {
+    downloadCtxBase: DataDownloadContextBase
+    firstYColDef?: OwidColumnDef
+}) => {
+    const [onlyVisible, setOnlyVisible] = useState(false)
+    const [shortColNames, setShortColNames] = useState(false)
+
+    const exLongName = props.firstYColDef?.name
+    const exShortName = props.firstYColDef?.shortName
+
+    // Some charts, like pre-ETL ones or csv-based explorers, don't have short names available for their variables
+    const shortNamesAvailable = !!exShortName
+
+    const downloadCtx: DataDownloadContextServerSide = useMemo(
+        () => ({
+            ...props.downloadCtxBase,
+            csvDownloadType: onlyVisible
+                ? CsvDownloadType.CurrentSelection
+                : CsvDownloadType.Full,
+            shortColNames,
+        }),
+        [props.downloadCtxBase, onlyVisible, shortColNames]
+    )
+
+    const csvUrl = useMemo(
+        () => getDownloadUrl("csv", downloadCtx),
+        [downloadCtx]
+    )
+    const metadataUrl = useMemo(
+        () => getDownloadUrl("metadata.json", downloadCtx),
+        [downloadCtx]
+    )
+
+    return (
+        <>
+            <div className="download-modal__data-section">
+                <div className="download-modal__heading-with-caption">
+                    <h3 className="grapher_h3-semibold">Data API</h3>
+                    <p className="grapher_label-2-regular">
+                        Use these URLs for programmatic use of this chart's
+                        data, and configure what you want to get using the
+                        options below. See also the code examples below.
```

Comment:
```suggestion
                        Use these URLs to programmatically access this chart's
                        data and configure your requests with the options below.{" "}
                        <a href="http://127.0.0.1:8000/projects/etl/api/#chart-data-api">
                            Our documentation
                        </a>{" "}
                        provides more information on how to use the API and you
                        can find a few code examples below.
```

I think this could be a good place to link to our docs, something like this. What do you think? The link styling would maybe have to be tweaked a little.